cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"encoding/json"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"path"
+	"strconv"
+	"strings"
+)
+
+func npmExecuteLint(config npmExecuteLintOptions, telemetryData *telemetry.CustomData) {
+	utils := npm.NpmUtilsBundle{}
+
+	err := runNpmExecuteLint(&utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runNpmExecuteLint(utils npm.NpmUtils) error {
+	packageJSONFiles, err := npm.FindPackageJSONFiles(utils)
+	if err != nil {
+		return err
+	}
+
+	packagesWithCilint, err := findPackagesWithCilint(packageJSONFiles, utils)
+	if err != nil {
+		return err
+	}
+
+	if len(packagesWithCilint) > 0 {
+		err = runCiLint(packagesWithCilint, utils)
+		if err != nil {
+			return err
+		}
+	} else {
+		err = runDefaultLint(utils)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func findPackagesWithCilint(packageJSONFiles []string, utils npm.NpmUtils) ([]string, error) {
+	var packageJSON map[string]interface{}
+	var packagesWithCilint []string
+
+	for _, file := range packageJSONFiles {
+		packageRaw, err := utils.FileRead(file)
+		if err != nil {
+			return nil, err
```

Comment:
Probably better to wrap the error to provide context:
```suggestion
			return nil, fmt.Errorf("failed to read %s to check for existence of ci-lint script: %w", file, err)
```
But maybe this shouldn't fail the step? This file will then just not be one of the files where linting is executed. It could be a warning in the log:
```suggestion
			log.Entry().Warnf("failed to read %s to check for existence of ci-lint script: %v", file, err)
			continue
```
(Since this was the only place where it could return an error, the function wouldn't need to return one then.)